src/python/pants/core/util_rules/partitions.py

```diff
@@ -36,7 +36,42 @@ class PartitionerType(Enum):
     """The plugin author has a rule to go from `RequestType.PartitionRequest` -> `Partitions`."""
 
     DEFAULT_SINGLE_PARTITION = "default_single_partition"
-    """Registers a partitioner which returns the inputs as a single partition."""
+    """Registers a partitioner which returns the inputs as a single partition
+
+    The returned partition will have no metadata.
+    """
+
+    DEFAULT_ONE_PARTITION_PER_INPUT = "default_one_partition_per_input"
+    """Registers a partitioner which returns a single-element partition per input.
+
+    Each of the returned partitions will have no metadata.
+    """
+
+    def default_rules(self, cls, *, by_file: bool) -> Iterable:
+        """Return an iterable of rules defining the default partitioning logic for this
+        `PartitionerType`.
+
+        NOTE: Not all `PartitionerType`s have default logic, so this method can return an empty iterable.
+
+        :param by_file: If `True`, rules returned from this method (if any) will compute partitions with
+          `str`-type elements, where each `str` value is the path to the resolved source field. If `False`, rule will compute
+          partitions with `FieldSet`-type elements.
+        """
+        if self == PartitionerType.CUSTOM:
+            # No default rules.
+            return
+        elif self == PartitionerType.DEFAULT_SINGLE_PARTITION:
+            rules_generator = (
+                _single_partition_file_rules if by_file else _single_partition_field_set_rules
+            )
+            yield from rules_generator(cls)
```

Comment:
I had a fever dream that we actually just have one rule for all the implementations that have closures for the `by_file`/single-v-multi partitions.

Thoughts?

Reply:
Maybe as a follow-up PR 😛 looking to get the test-batching work unblocked

Reply:
Yeah I didn't unset auto-merge because of that. Wanna file an issue real quick?